Modules/md5module.c

```diff
@@ -318,22 +318,31 @@ md5_done(struct md5_state *md5, unsigned char *out)
  * ------------------------------------------------------------------------
  */
 
-static PyTypeObject MD5type;
+typedef struct {
+    PyTypeObject* md5_type;
+} MD5State;
 
+static inline MD5State* md5_get_state(PyObject *module)
+{
+    void *state = PyModule_GetState(module);
+    assert(state != NULL);
+    return (MD5State *)state;
+}
 
 static MD5object *
-newMD5object(void)
+newMD5object(MD5State * st)
 {
-    return (MD5object *)PyObject_New(MD5object, &MD5type);
+    return (MD5object *)PyObject_New(MD5object, st->md5_type);
 }
 
-
 /* Internal methods for a hash object */
 
 static void
 MD5_dealloc(PyObject *ptr)
 {
+    PyObject *tp = Py_TYPE(ptr);
```

Comment:
Py_TYPE retunrs a `PyTypeObject *`:

```suggestion
    PyTypeObject *tp = Py_TYPE(ptr);
```

Same remark for other files.